src/dsmr/reader.h

```diff
@@ -69,10 +69,12 @@ class P1Reader {
      * output, the Stream is assumed to be already set up (e.g. baud
      * rate configured).
      */
-    P1Reader(Stream *stream, uint8_t req_pin)
+    P1Reader(Stream* stream, gpio_num_t req_pin = GPIO_NUM_NC)
```

Comment:
Where is `GPIO_NUM_NC` (and also `gpio_num_t`) defined? I think these are not standard Arduino API, so this will likely only work on whatever platform you found them. I am not sure if there is a completely portable "invalid pin" value, but I would suggest using 0xff, which seems unlikely enough to be a valid pin number. Maybe also good to define a constant for that, so maybe add `static const uint8 NO_PIN = 0xff` in the `P1Reader` class for this).